### ISSUE TITLE
`no-top-level-side-effects` should allow files whose only exports are `export type`

### ISSUE THREAD

**Issue by andersk** [sindresorhus/eslint-plugin-unicorn#3402]
### Description

`import type` and `export type` are removed when converting TypeScript to JavaScript, so `no-top-level-side-effects` should allow side effects in a file whose only exports are `export type`, just it already does in a file with no exports.

```diff
--- a/rules/no-top-level-side-effects.js
+++ b/rules/no-top-level-side-effects.js
@@ -12,7 +12,7 @@ const exportDeclarationTypes = new Set([
 	'ExportNamedDeclaration',
 ]);
 
-const isExportDeclaration = node => exportDeclarationTypes.has(node.type);
+const isExportDeclaration = node => exportDeclarationTypes.has(node.type) && node.exportKind !== 'type';
 
 const isAllowedAssignment = node => unwrapTypeScriptExpression(node).type === 'AssignmentExpression';
 ```

### Examples

```ts
// ❌
export const value = 1;
init();

// ✅
export type Foo = string;
init();
```
